straxen/plugins/nveto_recorder.py

```diff
@@ -0,0 +1,352 @@
+            assert rr['time'] < et[-1], 'This is odd this record omitted the current intervals.'
+
+        # Check if record start is in interval:
+        m_starts = rr['time'] >= st
+        # <= in m_ends is not ambiguous here since if start and end time of an interval would be the same
+        # they would have been merged into a single interval in coincidence.
+        m_ends = rr['time'] <= et
+
+        # Check if record end is in interval:
+        m_starts = m_starts | (strax.endtime(rr) >= st)
+        m_ends = m_ends | (strax.endtime(rr) <= et)
+        m = m_starts & m_ends
+
+        if np.any(m):
+            # This record is inside one of the interval
+            result[ind] = True
+
+            # Update intervals which we have seen already:
+            # If we have a funny record for which the start time is in interval 0
+            # and the end time in interval 1 we still set last interval seen to be
+            # the first interval. It might happen that this record is followed by
+            # a second record which is shorter falling only into interval 1. While
+            # there will be guaranteed by the definition of our coincidence another
+            # record at the start of the interval 1 which will increment last_interval_seen
+            last_interval_seen = np.argwhere(m)[0, 0] + last_interval_seen
+
+            # Now we have to get all associated records and set them to true:
+            for neighbors in record_links:
+                n = 0  # some trial counter
+                ri = ind
+                while n <= 1000:
+                    ri = neighbors[ri]
+                    if ri == -1:
+                        break
+                    else:
+                        result[ri] = True
+
+                if n == 1000:
+                    raise RuntimeWarning('Tried more than 1000 times to find'
+                                         ' neighboring record. This is odd.')
+    return result
+
+
+@export
+def coincidence(records, nfold=4, resolving_time=300):
+    """
+    Checks if n-neighboring events are less apart from each other then
+    the specified resolving time.
+
+    :param records: Can be anything which is of the dtype
+        strax.interval_dtype e.g. records, hits, peaks...
+    :param nfold: coincidence level.
+    :param resolving_time: Time window of the coincidence [ns].
+    :return: array containing the start times and end times of the
+            corresponding intervals.
+
+    Note:
+        The coincidence window is self-extending. If start times of two
+         intervals are exactly resolving_time apart from each other
+         they will be merged into a single interval.
+    """
+    if len(records):
+        start_times = _coincidence(records, nfold, resolving_time)
+        intervals = _merge_intervals(start_times, resolving_time)
+    else:
+        intervals = np.zeros((0, 2), np.int64)
+    return intervals
+
+
+def _coincidence(rr, nfold=4, resolving_time=300):
```

Comment:
In peaks we also have a tight coincidence function which looks somewhat simpler. I cannot see why/if it cannot work for you?

I know that if you do convolutions things can become quite slow. Have you tested that this is sufficently fast>

Reply:
I looked into tight coincidence and had the impression that it is very different from what I am doing here.